src/components/Collections/components/ContestCollections/CollectionItemNSFWLevelSelector.tsx

```diff
@@ -0,0 +1,57 @@
+import { SegmentedControl } from '@mantine/core';
+import { useCollection, useMutateCollection } from '~/components/Collections/collection.utils';
+import { browsingLevelLabels, browsingLevels } from '~/shared/constants/browsingLevel.constants';
+
+export const CollectionItemNSFWLevelSelector = ({
+  collectionId,
+  collectionItemId,
+  nsfwLevel,
+  onNsfwLevelUpdated,
+}: {
+  collectionId: number;
+  collectionItemId: number;
+  nsfwLevel?: number;
+  onNsfwLevelUpdated?: (value: string) => void;
+}) => {
+  const { collection, permissions, isLoading: loadingCollection } = useCollection(collectionId);
+  const {
+    updateCollectionItemNsfwLevel,
+    updatingCollectionItemNsfwLevel,
+    updateCollectionItemNsfwLevelPayload,
+  } = useMutateCollection();
+
+  const handleNSFWLevelChange = (value: string) => {
+    updateCollectionItemNsfwLevel(
+      {
+        collectionItemId,
+        nsfwLevel: parseInt(value, 10),
+      },
+      {
+        onSuccess: async () => {
+          onNsfwLevelUpdated?.(value);
+        },
+      }
+    );
```

Comment:
Probably a good idea to catch and return null, just so this promise doesn't fail silently
```suggestion
    ).catch(() => null); // Error handled at hook level
```